### PR TITLE
rgw_sal_motr : [CORTX-31446] Copy-object: send keep alive to client to avoid req timeout

### PR DESCRIPTION
**Pull request by pranavPawar07**
…
Ticket: https://jts.seagate.com/browse/CORTX-31446

Problem Description:
Timeout error for large-sized(more than 3GB) copy-object operation.

Problem Solution:
Call the RGWCopyObj::progress_cb after writing every part, which sends the partial response to the client so that it does not time out.

Signed-off-by: Pranav Diliprao Pawar <pranav.pawar@seagate.com>

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by sjain09 — CHANGES_REQUESTED** [review]
One comment

File: src/rgw/rgw_sal_motr.cc [hunk 1]
```diff
@@ -2205,6 +2209,10 @@ int MotrObject::copy_object_same_zone(RGWObjectCtx& obj_ctx,
   ldpp_dout(dpp, 20) << "Src Object Name : " << this->get_key().get_oid() << dendl;
   ldpp_dout(dpp, 20) << "Dest Object Name : " << dest_object->get_key().get_oid() << dendl;
 
+  //similar src and dest object name is not supported as of now
+  if(this->get_key().get_oid() == dest_object->get_key().get_oid())
```
[the remark this answers is not in the document]
Comment (pranavPawar07) [reply src/rgw/rgw_sal_motr.cc]: Done. please check

**Review by sjain09 — APPROVED** [review]
LGTM

**Review by DPG17 — APPROVED** [review]

File: src/rgw/rgw_sal_motr.cc [hunk 2]
```diff
@@ -2192,6 +2193,9 @@ int MotrCopyObj_CB::handle_data(bufferlist& bl, off_t bl_ofs, off_t bl_len)
                           write_offset << "failed rc=" << rc << dendl;
     }
     write_offset += bl_len;
+    if(progress_CB){
```
Comment (andriytk) [inline src/rgw/rgw_sal_motr.cc:2196]: Style: should be `if (..) {`.

**Pull request merged**